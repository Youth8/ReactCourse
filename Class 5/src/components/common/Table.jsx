import React from "react";

const Row = props => {
  const { row, head } = props;
  const Td = head ? "th" : "td";
  return (
    <tr>
      {row.map((cell, i) => {
        return <Td key={i}>{cell}</Td>;
      })}
    </tr>
  );
};

const Table = props => {
  const { head, body, title } = props;

  return (
    <div class="fds-data-table-wrapper">
      <div aria-hidden="true" class="fds-data-table-header">
        {title}
      </div>
      <div class="fds-data-table-scroll-wrapper">
        <table class="fds-data-table">
          <thead>
            <Row row={head} head />
          </thead>
          <tbody>
            {body.map((row, i) => (
              <Row key={row.key || i} row={row.cells || row} />
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
};

export default Table;
