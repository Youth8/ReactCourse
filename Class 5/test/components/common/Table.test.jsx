import React from "react";
import Table from "../../../src/components/common/Table";
import { shallow } from "enzyme";

describe("Tests for Table", () => {
  test("render empty table correctly", () => {
    const table = shallow(<Table head={[]} body={[]} />);
    expect(table).toMatchSnapshot();
  });

  test("render table correctly", () => {
    const head = ["first_name", "last_name"];
    const body = [{ first_name: "Michael", last_name: "Liu" }];
    const table = shallow(<Table head={head} body={body} />);
    expect(table).toMatchSnapshot();
  });
});
