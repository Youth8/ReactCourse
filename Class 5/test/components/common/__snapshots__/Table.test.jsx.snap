// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Table render empty table correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    body={Array []}
    head={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          aria-hidden="true"
          class="fds-data-table-header"
        />,
        <div
          class="fds-data-table-scroll-wrapper"
        >
          <table
            class="fds-data-table"
          >
            <thead>
              <Row
                head={true}
                row={Array []}
              />
            </thead>
            <tbody />
          </table>
        </div>,
      ],
      "class": "fds-data-table-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": undefined,
          "class": "fds-data-table-header",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <table
            class="fds-data-table"
          >
            <thead>
              <Row
                head={true}
                row={Array []}
              />
            </thead>
            <tbody />
          </table>,
          "class": "fds-data-table-scroll-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <Row
                  head={true}
                  row={Array []}
                />
              </thead>,
              <tbody />,
            ],
            "class": "fds-data-table",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Row
                  head={true}
                  row={Array []}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "head": true,
                  "row": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
              },
              "ref": null,
              "rendered": Array [],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            aria-hidden="true"
            class="fds-data-table-header"
          />,
          <div
            class="fds-data-table-scroll-wrapper"
          >
            <table
              class="fds-data-table"
            >
              <thead>
                <Row
                  head={true}
                  row={Array []}
                />
              </thead>
              <tbody />
            </table>
          </div>,
        ],
        "class": "fds-data-table-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "children": undefined,
            "class": "fds-data-table-header",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <table
              class="fds-data-table"
            >
              <thead>
                <Row
                  head={true}
                  row={Array []}
                />
              </thead>
              <tbody />
            </table>,
            "class": "fds-data-table-scroll-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <Row
                    head={true}
                    row={Array []}
                  />
                </thead>,
                <tbody />,
              ],
              "class": "fds-data-table",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Row
                    head={true}
                    row={Array []}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "head": true,
                    "row": Array [],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Tests for Table render table correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    body={
      Array [
        Object {
          "first_name": "Michael",
          "last_name": "Liu",
        },
      ]
    }
    head={
      Array [
        "first_name",
        "last_name",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          aria-hidden="true"
          class="fds-data-table-header"
        />,
        <div
          class="fds-data-table-scroll-wrapper"
        >
          <table
            class="fds-data-table"
          >
            <thead>
              <Row
                head={true}
                row={
                  Array [
                    "first_name",
                    "last_name",
                  ]
                }
              />
            </thead>
            <tbody>
              <Row
                row={
                  Object {
                    "first_name": "Michael",
                    "last_name": "Liu",
                  }
                }
              />
            </tbody>
          </table>
        </div>,
      ],
      "class": "fds-data-table-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": undefined,
          "class": "fds-data-table-header",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <table
            class="fds-data-table"
          >
            <thead>
              <Row
                head={true}
                row={
                  Array [
                    "first_name",
                    "last_name",
                  ]
                }
              />
            </thead>
            <tbody>
              <Row
                row={
                  Object {
                    "first_name": "Michael",
                    "last_name": "Liu",
                  }
                }
              />
            </tbody>
          </table>,
          "class": "fds-data-table-scroll-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <Row
                  head={true}
                  row={
                    Array [
                      "first_name",
                      "last_name",
                    ]
                  }
                />
              </thead>,
              <tbody>
                <Row
                  row={
                    Object {
                      "first_name": "Michael",
                      "last_name": "Liu",
                    }
                  }
                />
              </tbody>,
            ],
            "class": "fds-data-table",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Row
                  head={true}
                  row={
                    Array [
                      "first_name",
                      "last_name",
                    ]
                  }
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "head": true,
                  "row": Array [
                    "first_name",
                    "last_name",
                  ],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Row
                    row={
                      Object {
                        "first_name": "Michael",
                        "last_name": "Liu",
                      }
                    }
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "function",
                  "props": Object {
                    "row": Object {
                      "first_name": "Michael",
                      "last_name": "Liu",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            aria-hidden="true"
            class="fds-data-table-header"
          />,
          <div
            class="fds-data-table-scroll-wrapper"
          >
            <table
              class="fds-data-table"
            >
              <thead>
                <Row
                  head={true}
                  row={
                    Array [
                      "first_name",
                      "last_name",
                    ]
                  }
                />
              </thead>
              <tbody>
                <Row
                  row={
                    Object {
                      "first_name": "Michael",
                      "last_name": "Liu",
                    }
                  }
                />
              </tbody>
            </table>
          </div>,
        ],
        "class": "fds-data-table-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-hidden": "true",
            "children": undefined,
            "class": "fds-data-table-header",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <table
              class="fds-data-table"
            >
              <thead>
                <Row
                  head={true}
                  row={
                    Array [
                      "first_name",
                      "last_name",
                    ]
                  }
                />
              </thead>
              <tbody>
                <Row
                  row={
                    Object {
                      "first_name": "Michael",
                      "last_name": "Liu",
                    }
                  }
                />
              </tbody>
            </table>,
            "class": "fds-data-table-scroll-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <Row
                    head={true}
                    row={
                      Array [
                        "first_name",
                        "last_name",
                      ]
                    }
                  />
                </thead>,
                <tbody>
                  <Row
                    row={
                      Object {
                        "first_name": "Michael",
                        "last_name": "Liu",
                      }
                    }
                  />
                </tbody>,
              ],
              "class": "fds-data-table",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Row
                    head={true}
                    row={
                      Array [
                        "first_name",
                        "last_name",
                      ]
                    }
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "head": true,
                    "row": Array [
                      "first_name",
                      "last_name",
                    ],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Row
                      row={
                        Object {
                          "first_name": "Michael",
                          "last_name": "Liu",
                        }
                      }
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "function",
                    "props": Object {
                      "row": Object {
                        "first_name": "Michael",
                        "last_name": "Liu",
                      },
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
